lib/web_ui/dev/driver_manager.dart

```diff
@@ -20,7 +20,34 @@ import 'utils.dart';
 ///
 /// This manager can be used for both macOS and Linux.
 class ChromeDriverManager extends DriverManager {
-  ChromeDriverManager(String browser) : super(browser);
+
+  /// Directory which contains the preferred version of Chrome.
+  ///
+  /// On LUCI we are using the CIPD packages to control Chrome binaries we use.
+  /// There will be multiple CIPD packages loaded at the same time. Yaml file
+  /// `driver_version.yaml` contains the version number we want to use.
+  ///
+  /// Local integration tests are still using the system Chrome. This should be
+  /// fixed in issue: https://github.com/flutter/flutter/issues/53179.
+  ///
+  /// Initialized to the current first to avoid the `Non-nullable` error.
+  io.Directory _browserDriverDirWithVersion = io.Directory.current;
+
+  ChromeDriverManager(String browser) : super(browser) {
+    final io.File lockFile = io.File(pathlib.join(
+        environment.webUiRootDir.path, 'dev', 'driver_version.yaml'));
+    YamlMap _configuration = loadYaml(lockFile.readAsStringSync()) as YamlMap;
+    final String preferredChromeDriverVersion =
+        _configuration['preferred_version']['chrome'] as String;
+    print('preferred_version $preferredChromeDriverVersion');
+    _browserDriverDirWithVersion =io.Directory(pathlib.join(
+            environment.webUiDartToolDir.path,
+            'drivers',
+            browser,
+            preferredChromeDriverVersion,
+            '${browser}driver-${io.Platform.operatingSystem.toString()}'));
```

Comment:
Why do we need to append the operating system name? Is it possible to have drivers for different operating systems on the same computer? If not, then we can remove this part of the path because the name of the browser is already encoded in the path on line 46.

Reply:
We can't change this name. I already created the cipd packages. This subdirectory is in the package.

Reply:
I actually copied the naming from Chrome. When you download a chrome from chromium downloads and unzip, you will see the first directory would be `chrome-linux` or `chrome-mac`. We are following that naming.